modules/cover-image-workflowoperation/src/main/java/org/opencastproject/workflow/handler/coverimage/CoverImageWorkflowOperationHandlerBase.java

```diff
@@ -164,12 +156,10 @@ public WorkflowOperationResult start(final WorkflowInstance workflowInstance, Jo
       coverImage.setMimeType(MimeTypes.PNG);
 
       // Add tags
-      final String targetTags = StringUtils.trimToNull(operation.getConfiguration(TARGET_TAGS));
-      if (targetTags != null) {
-        for (String tag : asList(targetTags)) {
-          logger.trace("Tagging image with '{}'", tag);
-          if (StringUtils.trimToNull(tag) != null)
-            coverImage.addTag(tag);
+      for (String tag : tagsAndFlavors.getTargetTags()) {
+        logger.trace("Tagging image with '{}'", tag);
+        if (StringUtils.trimToNull(tag) != null) {
```

Comment:
Your code already ensures that, doesn't it?
```suggestion
```

Reply:
It does. Removed the check.